Feature/update pandoc runner

SMHFandA opened the pull request:
Modified PandocRunner class for usage ParamsBuilder.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pllug/pandoc-slave/5)

<!-- Reviewable:end -->

SMHFandA committed:
Added support for -o (output) pandoc parameter.

SMHFandA committed:
1. Remove build parameters functionality from class.
2. Added new reloaded run() functions.

SMHFandA committed:
Added documentation.

SMHFandA committed:
1. Wrote documentation.
2. Modified PandocRunner for usage ParamsBuilder.

Guitarheroua commented on pandoc-slave/pandoc-slave/pandocrunner.cpp:
```diff
@@ -3,103 +3,98 @@
+PandocSlave::PandocRunner::PandocRunner(const QString &pandocExePath, QObject* parent):
     QObject(parent)
   , mContent("")
   , mError("")
```
Please, remove `mContent("")` and `mError("")` initialization, as they will be initialize automatically.

Guitarheroua commented on pandoc-slave/pandoc-slave/pandocrunner.cpp:
```diff
@@ -3,103 +3,98 @@
+PandocSlave::PandocRunner::PandocRunner(const QString &pandocExePath, const QStringList &params, QObject *parent) :
+     PandocRunner(pandocExePath, parent)
+{
+    mParams = params;
```
Please, move `mParams = params;` to initialization list.

SMHFandA committed:
1. Wrote tests for ParamsBuilder class.

SMHFandA committed:
2. Fix initialization list for PandocRunner class.

SMHFandA committed:
3. Fix Documentation.

SMHFandA committed:
Remove needless QString initialization from initialization list.

SMHFandA committed:
1. Modified tests for ParamsBuilder.
2. Adde c++14 configuration to tests project.

SMHFandA committed:
1. Added clear() method.
2. Added c++14 configuration to pandoc-slave project.

Guitarheroua commented:
Reviewed 6 of 11 files at r2.
Review status: 6 of 11 files reviewed at latest revision, 3 unresolved discussions.

---

_[pandoc-slave/pandoc-slave/paramsbuilder.cpp, line 75 [r2]](https://reviewable.io:443/reviews/pllug/pandoc-slave/5#-KOs9daf3gVadm_FZPW7:-KOs9daf3gVadm_FZPW8:b-jfqead) ([raw file](https://github.com/pllug/pandoc-slave/blob/27f926f8951133d63282fef1cb03c6561b3c1e31/pandoc-slave/pandoc-slave/paramsbuilder.cpp#L75)):_

> ``` cpp
> {
>     mParams.clear();
>     mError = "";
> ```

QString also has method clear, please use it instead of `= "";`.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pllug/pandoc-slave/5)_

<!-- Sent from Reviewable.io -->

SMHFandA committed:
1. Added raw() method for settup mParams from simple string.
2. Use QString.clear() instead 'variable = ""'.
3. Added support --template pandoc parameter.
4. Added support --metadata pandoc parameter.

SMHFandA committed:
1. Modified tests for ParamsBuilder class.
2. Added documentation for supported formats and parameters.
3. Added documentation for supported pandoc template engine.

SMHFandA committed:
Added input files for tests.

SMHFandA committed:
Modified documentation.

SMHFandA committed:
Wrote tests form PandocRunner class.

Guitarheroua commented:
Reviewed 5 of 11 files at r2, 9 of 9 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/pllug/pandoc-slave/5)_

<!-- Sent from Reviewable.io -->

Guitarheroua commented:
Approved

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/PLLUG/pandoc-slave/pull-request/5/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=PLLUG/pandoc-slave)
